target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-fritzrepeater-1200.dts

```diff
@@ -237,6 +241,34 @@
 	status = "okay";
 };
 
+&wifi0 {
+	status = "okay";
+	qcom,ath10k-calibration-variant = "AVM-FRITZRepeater-1200";
+};
+
+&wifi1 {
+	status = "okay";
+	qcom,ath10k-calibration-variant = "AVM-FRITZRepeater-1200";
+};
+
+&gmac {
+	status = "okay";
+};
+
+&switch {
+	status = "okay";
+
+	/delete-property/ psgmii-ethphy;
+};
+
+&swport1 {
+	status = "okay";
+
+	label = "lan";
+	phy-handle = <&ethphy>;
+	phy-mode = "rgmii-id";
+};
+
 &ethphy0 {
 	status = "disabled";
 };
```

Comment:
Removing lines 272 to 275 gets me a bit further (don't disable ethphy0)

- [    1.144970] NET: Registered protocol family 17
- [    1.148808] 8021q: 802.1Q VLAN Support v1.8
- [    1.152995] Registering SWP/SWPB emulation handler
- [    1.188567] qca8k-ipq4019 c000000.switch: configuring for fixed/internal link mode
- [    1.188763] qca8k-ipq4019 c000000.switch: interface 'rgmii-id' (9) on port 1 is not supported
- [    1.195227] qca8k-ipq4019 c000000.switch: Link is Up - 1Gbps/Full - flow control rx/tx
- [    1.287205] qca8k-ipq4019 c000000.switch: interface 'rgmii-id' (9) on port 1 is not supported
- [    1.287647] qca8k-ipq4019 c000000.switch lan (uninitialized): validation of rgmii-id with support 0000000,00000000,000062ef and advertisement 0000000,00000000,000062ef failed: -22
- [    1.296461] qca8k-ipq4019 c000000.switch lan (uninitialized): failed to connect to PHY: -EINVAL
- [    1.311208] qca8k-ipq4019 c000000.switch lan (uninitialized): error -22 setting up PHY for tree 0, switch 0, port 1
- [    1.320535] DSA: tree 0 setup

Reply:
It looks like you need to switch swport1 back to swport5 as well, and that should be fine.

Reply:
I was trying to follow  [qcom-ipq4029-aruba-glenmorangie.dtsi] in (https://github.com/openwrt/openwrt/pull/4721/files#diff-619dfd8bfc54ca01560fd270dbf277dae30f8c60c8c34b1b9be4eea226d20268). Both ethphy and swport are renumbered to "5". But it didn't work:

- [    1.141193] Registering SWP/SWPB emulation handler
- [    1.178197] qca8k-ipq4019 c000000.switch: configuring for fixed/internal link mode
- [    1.178410] qca8k-ipq4019 c000000.switch lan (uninitialized): failed to connect to PHY: -ENODEV
- [    1.184746] qca8k-ipq4019 c000000.switch lan (uninitialized): error -19 setting up PHY for tree 0, switch 0, port 5
- [    1.193370] qca8k-ipq4019 c000000.switch: Link is Up - 1Gbps/Full - flow control rx/tx
- [    1.204035] DSA: tree 0 setup

Reply:
Now in your place ethphy is pointing to?

Reply:
you should see qcom-ipq4029-aruba-glenmorangie.dtsi do delete `ethernet-phy@0` and do add `ethphy: ethernet-phy@5`  in essedma driver version, but this model do not do this, so do not just copy it.

I do think after applying this PR, change swport1 back to swport5 and delete disabled node should be enough.

Reply:
https://openwrt.org/docs/guide-user/network/dsa/converting-to-dsa

I know little about VLAN with DSA, but found it, maybe you can try

Reply:
eth0.2 should work according to https://www.youtube.com/watch?v=qeuZqRqH-ug. But maybe bridge vlan filtering should be activated while it wasn't necessary before.